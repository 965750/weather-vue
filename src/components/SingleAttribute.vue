<template>
    <div class="attribute mb-4 pa-0">
        <div class="attribute__value pt-1 text-center px-1">
            <font-awesome-icon
                :icon="icon"
            />
            <p class="my-0">{{ value }}</p>
        </div>
        <p class="attribute__label text-center px-1">
            {{ label }}
        </p>
    </div>
</template>

<script>
export default {
  name: 'SingleAttribute',
  props: {
    icon: {
      required: true,
      type: String,
    },
    label: {
      required: true,
      type: String,
    },
    value: {
      required: true,
      type: [String, Number],
    },
  },
};
</script>

<style lang="scss" scoped>
@import '../assets/main';

.attribute {
    background: lighten($midnight, 15%);
    border-radius: 10px;
    height: 60px;
    overflow: hidden;
    width: 23%;
    margin: 1%;

    &__value {
        line-height: 16px;
        font-size: 12px;
        height: 40px;
    }

    &__label {
        height: 20px;
        line-height: 20px;
        font-size: 11px;
        background: lighten($midnight, 5%);
    }
}
</style>
