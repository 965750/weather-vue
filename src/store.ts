import Vue from 'vue';
import Vuex from 'vuex';
import axios from 'axios';
import config from '@/config';

import mutations from './mutations';
import getters from './getters';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    lang: 'pl',
    card: null,
    error: null,
  },
  actions: {
    citySearch({ commit }, city: string) {
      axios
        .get(`https://api.openweathermap.org/data/2.5/forecast?q=${city}&cnt=${config.cnt}&APPID=${config.apiKey}`)
        .then((res) => {
          commit('setCard', res.data);
        })
        .catch((err) => {
          commit('setCard__Error', err);
        });
    },
  },
  mutations,
  getters,
});
