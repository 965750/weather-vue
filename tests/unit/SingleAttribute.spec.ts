import {
  shallowMount,
  createLocalVue,
} from '@vue/test-utils';
import VueI18n from 'vue-i18n';
import SingleAttribute from '@/components/SingleAttribute.vue';
import filters from '@/myFilters';
import store from '@/store';
import messages from '@/locale/languages';

const localVue = createLocalVue();
localVue.use(VueI18n);

const i18n = new VueI18n({
  locale: store.getters.getLanguage,
  messages,
});

filters.forEach((filter) => {
  localVue.filter(filter.name, filter.handler);
});

const initialProps = {
  icon: 'sun',
  label: 'clouds',
  value: 25,
};

describe('SingleAttribute.vue', () => {
  const createWrapper = (props = initialProps) => shallowMount(SingleAttribute, {
    i18n,
    localVue,
    propsData: {
      ...props,
    },
    stubs: {
      'font-awesome-icon': true,
    },
  });

  it('should match initial snapshot', () => {
    const wrapper = createWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });
});
