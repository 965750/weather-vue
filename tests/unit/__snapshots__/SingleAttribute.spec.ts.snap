// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleAttribute.vue should match initial snapshot 1`] = `
<div
  class="attribute mb-4 pa-0"
>
  <div
    class="attribute__value pt-1 text-center px-1"
  >
    <font-awesome-icon-stub
      icon="sun"
    />
     
    <p
      class="my-0"
    >
      25
    </p>
  </div>
   
  <p
    class="attribute__label text-center px-1"
  >
    
        chmury
    
  </p>
</div>
`;
